packages/client/src/utils/flattenObject.ts

```diff
@@ -0,0 +1,24 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+type NotObject<T> = Exclude<T, Record<string, any>>;
+type FlattenedObject<T> = Record<string, NotObject<T>>;
+
+export default function flattenObject<T>(
```

Comment:
Generic functions should use the type parameters more than once to relate values. In this case it is being used only in one place, so it is not providing value for the consumers. I believe the correct function definition should be quite a bit more involved (like defining another type on the fly that will add recursively all properties that are of type object to the passed type).

Let's simplify this a bit and remove the generic definition and return a Record<string, unknown>. Also the `object` parameter could be renamed to sourceObj or just source and could be of type Record<string, unknown> as well.